idea-plugin/src/main/kotlin/org/jetbrains/compose/desktop/ide/preview/ui/PreviewPanel.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2020-2022 JetBrains s.r.o. and respective authors and developers.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package org.jetbrains.compose.desktop.ide.preview.ui
+
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.project.Project
+import com.intellij.ui.SimpleTextAttributes
+import com.intellij.ui.components.JBPanel
+import com.intellij.util.ui.StatusText
+import java.awt.Color
+import java.awt.Dimension
+import java.awt.Graphics
+import java.awt.image.BufferedImage
+import java.util.concurrent.atomic.AtomicReference
+import javax.swing.SwingUtilities
+
+internal class PreviewPanel(private val myProject: Project) : JBPanel<PreviewPanel>() {
+    sealed class PreviewPanelState {
+        data class Image(val image: BufferedImage, val dimension: Dimension) : PreviewPanelState()
+        class Error(val error: String) : PreviewPanelState()
+    }
+    private val myState = AtomicReference<PreviewPanelState>()
+    private val myStatusText = object : StatusText(this) {
+        override fun isStatusVisible(): Boolean {
+            return myState.get() is PreviewPanelState.Error
+        }
+    }
+
+    init {
+        SwingUtilities.invokeLater {
```

Comment:
Shall we do that on some particular event AWT init event instead to make sure everything is properly inited?

Reply:
Honestly I have no idea :)
But it seems that `createToolWindowContent` in which we create PreviewPanel is called after Intellij is fully initialized, so I would assume it is not necessary.